polynote-frontend/polynote/messages.js

```diff
@@ -210,19 +210,143 @@ export class CellResult extends Message {
 
 CellResult.codec = combined(shortStr, tinyStr, Result.codec).to(CellResult);
 
+
 export class ContentEdit {
+
+    // TODO: starting to think overhead of scala.js would have been worth it to avoid duplicating all this logic...
```

Comment:
This is the heart of the OT.

Reply:
(on the client side)

Reply:
+1 to your comment on scala.js - this sort of thing looks like the perfect use-case of scala.js IMO. 

In the meantime, we may want to see whether we could at least have the same sort of test coverage that we have on the Scala side here - are you familiar with any javascript testing frameworks? (I'm not...)

Reply:
No, I'm not either.

I'll play around with the idea of using scala.js for the messaging and whatnot. Not sure about it though, since it probably doesn't handle buffers the fast way (those are relatively new APIs in JS). Anyway we'll see.